### PR TITLE
[CBRD-23845] Support SYNONYM

### PR DESCRIPTION
**Pull request by youngjinj**
http://jira.cubrid.org/browse/CBRD-23845

A table that is not owned by itself can be used only by using the name as owner_name.table_name. It is difficult to use a table by checking the owner name every time. So it supports synonyms. Synonym can be used as an alias for the name owner_name.table_name.

### CODE REVIEW

File: src/parser/csql_grammar.y [hunk 1]
```diff
@@ -3111,6 +3116,40 @@ create_stmt
...
+	  	{ push_msg (MSGCAT_SYNTAX_SYNONYM_INVALID_CREATE); }	/* 3 */
+	  opt_access_modifier		/* 4 */
+	  SYNONYM			/* 5 */
+	  synonym_name_without_dot	/* 6 */
+	  For				/* 7 */
+	  class_name			/* 8 */
+	  opt_comment_spec		/* 9 */
+	  	{ pop_msg(); }
+		{{
+
```
Comment (ctshim) [inline src/parser/csql_grammar.y:3131]: Please add below
`DBG_TRACE_GRAMMAR(create_stmt, | CREATE  opt_or_replace opt_access_modifier SYNONYM synonym_name_without_dot For class_name opt_comment_spec);`

File: src/parser/csql_grammar.y [hunk 2]
```diff
@@ -3928,6 +3967,38 @@ alter_stmt
...
+		{ push_msg (MSGCAT_SYNTAX_SYNONYM_INVALID_ALTER); }	/* 2 */
+	  opt_access_modifier		/* 3 */
+	  SYNONYM			/* 4 */
+	  synonym_name			/* 5 */
+	  For				/* 6 */
+	  class_name			/* 7 */
+	  opt_comment_spec		/* 8 */
+		{ pop_msg(); }
+		{{
+
```
Comment (ctshim) [inline src/parser/csql_grammar.y:3981]: Please add below
`DBG_TRACE_GRAMMAR(alter_stmt, | ALTER opt_access_modifier SYNONYM synonym_name For class_name opt_comment_spec);`

File: src/parser/csql_grammar.y [hunk 3]
```diff
@@ -4020,6 +4091,35 @@ rename_stmt
...
+	| RENAME			/* 1 */
+		{ push_msg (MSGCAT_SYNTAX_SYNONYM_INVALID_RENAME); }	/* 2 */
+	  opt_access_modifier		/* 3 */
+	  SYNONYM			/* 4 */
+	  synonym_name			/* 5 */
+	  as_or_to			/* 6 */
+	  synonym_name_without_dot	/* 7 */
+	  	{ pop_msg(); }
+		{{
+
```
Comment (ctshim) [inline src/parser/csql_grammar.y:4106]: Please add below
`DBG_TRACE_GRAMMAR(rename_stmt, | RENAME opt_access_modifier SYNONYM synonym_name as_or_to synonym_name_without_dot);`

File: src/parser/csql_grammar.y [hunk 4]
```diff
@@ -4343,7 +4443,37 @@ drop_stmt
...
+		DBG_PRINT}}
+	| DROP				/* 1 */
+		{ push_msg (MSGCAT_SYNTAX_SYNONYM_INVALID_DROP); }	/* 2 */
+	  opt_access_modifier		/* 3 */
+	  SYNONYM			/* 4 */
+	  opt_if_exists			/* 5 */
+	  synonym_name			/* 6 */
+	  	{ pop_msg(); }
+		{{
+
```
Comment (ctshim) [inline src/parser/csql_grammar.y:4455]: Please add below
`DBG_TRACE_GRAMMAR(drop_stmt, | DROP SERVER opt_access_modifier SYNONYM opt_if_exists  synonym_name );`

File: src/parser/csql_grammar.y [hunk 5]
```diff
@@ -5428,6 +5558,16 @@ serial_name
 		{ $$ = $1; }
 	;
 
+synonym_name_without_dot
+	: user_specified_name_without_dot
+		{ $$ = $1; }
```
Comment (ctshim) [inline src/parser/csql_grammar.y]: Please add below
```
{  DBG_TRACE_GRAMMAR(synonym_name_without_dot, : user_specified_name_without_dot );
   $$ = $1;
}
```

File: src/parser/csql_grammar.y [hunk 6]
```diff
@@ -5428,6 +5558,16 @@ serial_name
...
+	;
+
+synonym_name
+	: user_specified_name
+		{ $$ = $1; }
```
Comment (ctshim) [inline src/parser/csql_grammar.y]: Please add below
```
{  DBG_TRACE_GRAMMAR(synonym_name, : user_specified_name );
   $$ = $1;
}
```

**Review by ctshim — APPROVED** [review]

File: src/parser/csql_grammar.y (hunk 5 above)
Comment (youngjinj) [reply src/parser/csql_grammar.y]: Thank you. Added.

File: src/parser/csql_grammar.y (hunk 6 above)
Comment (youngjinj) [reply src/parser/csql_grammar.y]: Thank you. Added.

File: src/parser/csql_grammar.y (hunk 4 above)
Comment (youngjinj) [reply src/parser/csql_grammar.y:4455]: Thank you. Added.

File: src/parser/csql_grammar.y (hunk 3 above)
Comment (youngjinj) [reply src/parser/csql_grammar.y:4106]: Thank you. Added.

File: src/parser/csql_grammar.y (hunk 2 above)
Comment (youngjinj) [reply src/parser/csql_grammar.y:3981]: Thank you. Added.

File: src/parser/csql_grammar.y (hunk 1 above)
Comment (youngjinj) [reply src/parser/csql_grammar.y:3131]: Thank you. Added.

**Pull request merged**